js/poloniexfutures.js

```diff
@@ -0,0 +1,920 @@
+        const postOnly = this.safeValue (params, 'postOnly', false);
+        const hidden = this.safeValue (params, 'hidden');
+        if (postOnly && (hidden !== undefined)) {
+            throw new BadRequest (this.id + ' createOrder() does not support the postOnly parameter together with a hidden parameter');
+        }
+        const iceberg = this.safeValue (params, 'iceberg');
+        if (iceberg) {
+            const visibleSize = this.safeValue (params, 'visibleSize');
+            if (visibleSize === undefined) {
+                throw new ArgumentsRequired (this.id + ' createOrder() requires a visibleSize parameter for iceberg orders');
+            }
+        }
+        params = this.omit (params, [ 'timeInForce', 'stopPrice', 'triggerPrice' ]); // Time in force only valid for limit orders, exchange error when gtc for market orders
+        const response = await this.privatePostOrders (this.extend (request, params));
+        //
+        //    {
+        //        code: "200000",
+        //        data: {
+        //            orderId: "619717484f1d010001510cde",
+        //        },
+        //    }
+        //
+        const data = this.safeValue (response, 'data', {});
+        return {
+            'id': this.safeString (data, 'orderId'),
+            'clientOrderId': undefined,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'lastTradeTimestamp': undefined,
+            'symbol': undefined,
+            'type': undefined,
+            'side': undefined,
+            'price': undefined,
+            'amount': undefined,
+            'cost': undefined,
+            'average': undefined,
+            'filled': undefined,
+            'remaining': undefined,
+            'status': undefined,
+            'fee': undefined,
+            'trades': undefined,
+            'timeInForce': undefined,
+            'postOnly': undefined,
+            'stopPrice': undefined,
+            'info': response,
+        };
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api];
+        const versions = this.safeValue (this.options, 'versions', {});
+        const apiVersions = this.safeValue (versions, api, {});
+        const methodVersions = this.safeValue (apiVersions, method, {});
+        const defaultVersion = this.safeString (methodVersions, path, this.version);
+        const version = this.safeString (params, 'version', defaultVersion);
+        const tail = '/api/' + version + '/' + this.implodeParams (path, params);
+        url += tail;
+        const query = this.omit (params, path);
+        if (api === 'public') {
+            const queryLength = Object.keys (query).length;
+            if (queryLength) {
+                url += '?' + this.urlencode (query);
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            if (method !== 'GET' && method !== 'HEAD') {
```

Comment:
so this is not right, 
you should only url-encode if Get or Head, so your if should be something like:

```Javascript
            const bodyEncoded = this.urlencode (query);
            if (method !== 'GET' && method !== 'HEAD') {
                body = query;
            } else {
                url += '?' + bodyEncoded;
            }
```